ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -328,5 +344,120 @@ static void errorAndExit(String error) {
     System.err.println(error);
     System.exit(1);
   }
+
+  /**
+   * Is the rebalancer class {@link org.apache.helix.controller.rebalancer.DelayedAutoRebalancer}
+   * @param rebalancerClassName name of the rebalancer class.
+   * @return true if rebalancer is {@link org.apache.helix.controller.rebalancer.DelayedAutoRebalancer}, false otherwise.
+   */
+  private static boolean isDelayedRebalanceEnabled(String rebalancerClassName) {
+    return (rebalancerClassName != null) && rebalancerClassName.endsWith(DELAYED_REBALANCER_CLASS_NAME);
+  }
+
+  /**
+   * Class for ideal state configs.
+   */
+  static class IdealStateConfigFields {
+    private int numReplicas;
+    private String stateModelDefRef;
+    private String rebalanceStrategy;
+    private int minActiveReplicas;
+    private String rebalancerClassName;
+    private long rebalanceDelayInMins;
+
+    /**
+     * @return {@code numReplicas}
+     */
+    public int getNumReplicas() {
+      return numReplicas;
+    }
+
+    /**
+     * @return {@code stateModelDefRef}
+     */
+    public String getStateModelDefRef() {
+      return stateModelDefRef;
+    }
+
+    /**
+     * @return {@code rebalanceStrategy}
+     */
+    public String getRebalanceStrategy() {
+      return rebalanceStrategy;
+    }
+
+    /**
+     * @return {@code minActiveReplicas}
+     */
+    public int getMinActiveReplicas() {
+      return minActiveReplicas;
+    }
+
+    /**
+     * @return {@code rebalancerClassName}
+     */
+    public String getRebalancerClassName() {
+      return rebalancerClassName;
+    }
+
+    /**
+     * @return {@code rebalanceDelayInMins}
+     */
+    public long getRebalanceDelayInMins() {
+      return rebalanceDelayInMins;
+    }
+  }
+
+  /**
+   * Class for cluster config fields.
+   */
+  static class ClusterConfigFields {
+    private int maxOfflineInstancesAllowed;
+    private int numOfflineInstancesForAutoExit;
+    private boolean allowAutoJoin;
+
+    /**
+     * @return {@code maxOfflineInstancesAllowed}
+     */
+    public int getMaxOfflineInstancesAllowed() {
+      return maxOfflineInstancesAllowed;
+    }
+
+    /**
+     * @return {@code numOfflineInstancesForAutoExit}
+     */
+    public int getNumOfflineInstancesForAutoExit() {
+      return numOfflineInstancesForAutoExit;
+    }
+
+    /**
+     * @return {@code allowAutoJoin}
+     */
+    public boolean isAllowAutoJoin() {
+      return allowAutoJoin;
+    }
+  }
+
+  /**
+   * Class for configs passed to the {@link HelixVcrPopulateTool}
+   */
+  static class Config {
+    private IdealStateConfigFields idealStateConfigFields;
+    private ClusterConfigFields clusterConfigFields;
```

Comment:
I suppose there is ordering requirement (i.e. idealStateConfigFields comes first) when preparing the json file. In that case, it's a bit of error-prone for user to use this tool.

Reply:
There is no ordering requirement. The ordering of the keys in the json file doesn't matter, as long as the general structure is correct.